plugins/crd/plugin_impl_crd.go

```diff
@@ -0,0 +1,345 @@
+				if !reflect.DeepEqual(old, cur) {
+					networkservicechannelEnqueue(cur)
+				}
+			},
+			DeleteFunc: networkservicechannelEnqueue,
+		},
+	)
+
+	// Start the informer. This will cause it to begin receiving updates from
+	// the configured API server and firing event handlers in response.
+	sharedFactoryNSC.Start(stopCh)
+	plugin.Log.Info("Started NetworkServiceChannel informer factory.")
+
+	// Wait for the informer cache to finish performing it's initial sync of
+	// resources
+	if !cache.WaitForCacheSync(stopCh, informer.HasSynced) {
+		plugin.Log.Errorf("Error waiting for informer cache to sync: %s", err.Error())
+	}
+
+	plugin.Log.Info("NetworkServiceChannel Informer is ready")
+}
+
+func informerNetworkserviceendpoints(plugin *Plugin) {
+	var err error
+
+	if err != nil {
+		plugin.Log.Errorf("Error creating api client: %s", err.Error())
+	}
+
+	// We use a shared informer from the informer factory, to save calls to the
+	// API as we grow our application and so state is consistent between our
+	// control loops. We set a resync period of 30 seconds, in case any
+	// create/replace/update/delete operations are missed when watching
+	sharedFactoryNSE = factory.NewSharedInformerFactory(plugin.crdClient, time.Second*30)
+
+	informer := sharedFactoryNSE.Networkservice().V1().NetworkServiceEndpoints().Informer()
+	// we add a new event handler, watching for changes to API resources.
+	informer.AddEventHandler(
+		cache.ResourceEventHandlerFuncs{
+			AddFunc: networkserviceendpointEnqueue,
+			UpdateFunc: func(old, cur interface{}) {
+				if !reflect.DeepEqual(old, cur) {
+					networkserviceendpointEnqueue(cur)
+				}
+			},
+			DeleteFunc: networkserviceendpointEnqueue,
+		},
+	)
+
+	// Start the informer. This will cause it to begin receiving updates from
+	// the configured API server and firing event handlers in response.
+	sharedFactoryNSE.Start(stopCh)
+	plugin.Log.Info("Started NetworkServiceEndpoints informer factory.")
+
+	// Wait for the informer cache to finish performing it's initial sync of
+	// resources
+	if !cache.WaitForCacheSync(stopCh, informer.HasSynced) {
+		plugin.Log.Errorf("Error waiting for informer cache to sync: %s", err.Error())
+	}
+
+	plugin.Log.Info("NetworkServiceEndpoint Informer is ready")
+}
+
+// AfterInit This will create all of the CRDs for NetworkServiceMesh.
+func (plugin *Plugin) AfterInit() error {
+	var err error
+	var crdname string
+
+	// create clientset and create our CRD, this only needs to run once
+	plugin.apiclientset, err = apiextcs.NewForConfig(plugin.k8sClientConfig)
+	if err != nil {
+		panic(err.Error())
+	}
+
+	// Create an instance of our own API client
+	plugin.crdClient, err = client.NewForConfig(plugin.k8sClientConfig)
+
+	if err != nil {
+		plugin.Log.Errorf("Error creating CRD client: %s", err.Error())
+		panic(err.Error())
+	}
+
+	crdname = reflect.TypeOf(v1.NetworkServiceEndpoint{}).Name()
+	err = createCRD(plugin, v1.FullNSMEPName,
+		v1.NSMGroup,
```

Comment:
Experiment with passing in the struct vs. putting these "v1.*" items directly into createCRD(). Likely the latter.